test/server/updater/testabstractupdater.cpp

```diff
@@ -0,0 +1,95 @@
+static const std::string smallVersionJsonUpdateStr =
+    R"({"result":"success","data":{"application_id":27,"has_prod_next":false,"version":{"tag":"1.1.1","tag_updated_at":"2020-06-04 15:06:37","version_changelog":"test","type":"production","build_version":"20200604","build_min_os_version":"20200604","download_link":"test","data":["[]"]},"application":{"id":27,"name":"com.infomaniak.drive","platform":"mac-os","store":"kStore","api_id":"com.infomaniak.drive","min_version":"1.1.1","next_version_rate":0,"published_versions":[{"tag":"1.1.1","tag_updated_at":"2020-06-04 15:06:37","version_changelog":"test","type":"production","build_version":"20200604","build_min_os_version":"20200604","download_link":"test","data":["[]"]},{"tag":"1.1.1","tag_updated_at":"2020-06-04 15:06:12","version_changelog":"test","type":"beta","build_version":"20200604","build_min_os_version":"20200604","download_link":"test","data":["[]"]},{"tag":"1.1.1","tag_updated_at":"2020-06-04 15:05:44","version_changelog":"test","type":"internal","build_version":"20200604","build_min_os_version":"20200604","download_link":"test","data":["[]"]},{"tag":"1.1.1","tag_updated_at":"2020-06-04 15:03:29","version_changelog":"test","type":"production-next","build_version":"20200604","build_min_os_version":"20200604","download_link":"test","data":["[]"]}]}}})";
+
+void TestAbstractUpdater::setUp() {
+    LOG_DEBUG(Log::instance()->getLogger(), "$$$$$ Set Up");
+
+    // Create parmsDb
+    bool alreadyExists = false;
+    std::filesystem::path parmsDbPath = Db::makeDbName(alreadyExists, true);
+    ParmsDb::instance(parmsDbPath, "3.4.0", true, true);
+    ParametersCache::instance()->parameters().setExtendedLog(true);
+}
+
+void TestAbstractUpdater::tearDown() {}
+
+class TestGetAppVersionJob final : public GetAppVersionJob {
+    public:
+        TestGetAppVersionJob(const Platform platform, const std::string &appID, const bool updateAvailable)
+            : GetAppVersionJob(platform, appID), _updateAvailable(updateAvailable) {}
+
+        void runJob() noexcept override {
+            std::istringstream iss(_updateAvailable ? bigVersionJsonUpdateStr : smallVersionJsonUpdateStr);
+            std::istream is(iss.rdbuf());
+            GetAppVersionJob::handleResponse(is);
+        }
+
+    private:
+        bool _updateAvailable{false};
+};
+
+void TestAbstractUpdater::testCheckUpdateAvailable() {
+    const auto appUid = "1234567890";
+
+    // Version is higher than current version
+    auto *testJob = new TestGetAppVersionJob(CommonUtility::platform(), appUid, true);
```

Comment:
You can isolate each subtest with braces (like ´{ subtest }´) to make sure any of your temporary variables is misused by a subsequent test.